### PR TITLE
Fix issue with DROP view/table, dropping irrelevant user privileges

### PR DESCRIPTION
**Pull request by matriv**
Fixed the logic in `PrivilegesModifier` which caused all existing users and roles to loose their assigned privileges when a table or view is dropped.

Fixes: #16749

### CODE REVIEW

File: server/src/main/java/io/crate/role/PrivilegesModifier.java [hunk 1]
```diff
@@ -159,14 +159,14 @@ public static long dropTableOrViewPrivileges(Metadata.Builder mdBuilder,
...
-                if (ident.equals(tableOrViewIdent)) {
-                    affectedPrivileges++;
+                if (securable == Securable.TABLE || securable == Securable.VIEW) {
+                    String ident = subject.ident();
+                    assert ident != null : "ident must not be null for securable 'TABLE' or 'VIEW'";
+                    if (ident.equals(tableOrViewIdent)) {
+                        affectedPrivileges++;
+                    } else {
+                        updatedPrivileges.add(privilege);
+                    }
```
Comment (seut) [inline server/src/main/java/io/crate/role/PrivilegesModifier.java:169]: could be simplified, but maybe personal perference:
```suggestion
                String ident = subject.ident();
                if ((securable == Securable.TABLE || securable == Securable.VIEW)
                    && tableOrViewIdent.equals(ident)) {
                    affectedPrivileges++;
```

File: server/src/test/java/io/crate/role/metadata/UsersPrivilegesMetadataTest.java [hunk 2]
```diff
@@ -68,8 +69,13 @@ static UsersPrivilegesMetadata createMetadata() {
         }
         usersPrivileges.put(USER_WITHOUT_PRIVILEGES, new HashSet<>());
         usersPrivileges.put(USER_WITH_DENIED_DQL, new HashSet<>(Collections.singletonList(DENY_DQL)));
-        usersPrivileges.put(USER_WITH_SCHEMA_AND_TABLE_PRIVS, new HashSet<>(
-            Arrays.asList(GRANT_SCHEMA_DML, GRANT_TABLE_DQL, GRANT_TABLE_DDL, GRANT_VIEW_DQL, GRANT_VIEW_DML, GRANT_VIEW_DDL)));
+        usersPrivileges.put(USER_WITH_TABLE_VIEW_SCHEMA_AL_PRIVS, new HashSet<>(
```
Comment (seut) [inline server/src/test/java/io/crate/role/metadata/UsersPrivilegesMetadataTest.java:72]: How is this change related?

File: docs/appendices/release-notes/5.8.5.rst [hunk 3]
```diff
@@ -47,10 +47,17 @@ See the :ref:`version_5.8.0` release notes for a full list of changes in the
...
+  privileges, like: ``Administration Language (AL)``, when a table or a view
+  is dropped.
+
 - Improved the memory accounting for values of type ``geo_shape`` to avoid
   running into out of memory errors or higher than expected GC load when running
   queries like ``INSERT INTO ... (query)``.
 
+- Fixed an issue that would cause users to loose irrelevant privileges, like:
+  ``Administration Language (AL)``, when a table or a view is dropped.
+
```
Comment (BaurzhanSakhariev) [inline docs/appendices/release-notes/5.8.5.rst:60]: Duplicate entry?

File: server/src/test/java/io/crate/role/metadata/UsersPrivilegesMetadataTest.java (hunk 2 above)
Comment (matriv) [reply server/src/test/java/io/crate/role/metadata/UsersPrivilegesMetadataTest.java:72]: This test is about the old User Metadata, and it imports the same users from the other test. Since it's something to be removed in the future, I didn't spend time to separate them.

File: docs/appendices/release-notes/5.8.5.rst (hunk 3 above)
Comment (matriv) [reply docs/appendices/release-notes/5.8.5.rst:60]: yep, sorry, did something wrong when rebasing.

**Review by seut — APPROVED** [review]

**Review by BaurzhanSakhariev — APPROVED** [review]
